### ISSUE TITLE
Beginners problems with amiwrap ad installToolbox

### ISSUE THREAD

**Issue by thomassligon** [AMICI-developer/AMICI#2]
Here are my notes, sorry for the length. My workaround for the installToolbox is at the end.

What does “@” mean in “@amimodel”? According to the warning below, this might be a “method directory”, but I can’t find out what that is.

Does installToolbox.m need to run in AMICI directory?
addpath(genpath('auxiliary'))
Warning: Method directories not allowed in MATLAB path: @amimodel 
> In path (line 109)
  In addpath (line 86)
  In installToolbox (line 2)
  In MakeModels (line 4)

Error using amiwrap (line 23)
 symfun must be the name of a matlab function in the matlab path
Error in MakeModels (line 5)
    amiwrap('M1Trivial','M1Trivial');
“symfun” has no meaning for user. Documentation calls it “example_model_syms” Simply “parameter 2” would mean more.

Now trying
    amiwrap('M1Trivial','models/M1Trivial.m');

Error using eval
Undefined function or variable 'models'.
Error in amimodel (line 93)
                model = eval(symfun);
Error in amiwrap (line 55)
    model = amimodel(symfun,modelname);
Error in MakeModels (line 5)
    amiwrap('M1Trivial','models/M1Trivial.m');

Now I tried renaming “M1Trivial.m” to “M1Trivial_syms.m”.
Same problems.

Now trying:
    addpath('AMICI','AMICI/models');
    cd('AMICI');
    installToolbox;
    cd('models');
    model=M1Trivial_syms;
    cd('..');
    amiwrap('M1Trivial',model);
Error:
Error using amiwrap (line 20)
 symfun must be a string
Error in MakeModels (line 9)
    amiwrap('M1Trivial',model);
That disagrees with the document

Now I have put my make script in the models folder and I am getting much further. There is a problem with an undeclared identifier M_PI.
Error using mex
amiwrap.c
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\include/cvodewrap.h(107): warning C4098: 'AMIFree': 'void' function
returning a value
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(36): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(713): warning C4554: '|': check operator precedence for
possible error; use parentheses to clarify precedence
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(723): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(724): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(731): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(732): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(738): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(739): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(746): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(747): warning C4267: '=': conversion from 'size_t' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(916): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\src/amici.c(920): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(84): warning C4244: '=': conversion from 'double' to 'int',
possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(110): warning C4244: '=': conversion from 'double' to 'int',
possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(115): warning C4244: '=': conversion from 'double' to 'int',
possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(125): warning C4244: '=': conversion from 'double' to 'int',
possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(130): warning C4244: '=': conversion from 'double' to 'int',
possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(136): warning C4244: '=': conversion from 'double' to 'int',
possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(198): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(204): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(267): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(309): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(314): error C2065: 'M_PI': undeclared identifier
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(416): warning C4244: 'function': conversion from 'double' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(417): warning C4244: 'function': conversion from 'double' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(436): warning C4244: 'function': conversion from 'double' to
'int', possible loss of data
C:\Data\Tom\Research\Raedler\mRNA-Helmholtz\Tom\AMICI\amiwrap.c(437): warning C4244: 'function': conversion from 'double' to
'int', possible loss of data
Error in amimodel/compileC (line 381)
            eval(['mex ' COPT ' ' CLIBS ' -output ' fullfile(wrap_path,'models',this.modelname,[prefix '_' this.modelname])
            ' ' fullfile(wrap_path,cstr)  includesstr objectsstr ])
Error in amiwrap (line 76)
    model.compileC();
Error in MakeModels (line 13)
    amiwrap('M1Trivial','M1Trivial');

Workaround for installToolbox:
    modelPath=fileparts(mfilename('fullpath'));
    cd('..');
    amiciPath=pwd;
    addpath(amiciPath,modelPath);
%     installToolbox;
    addpath(fullfile(amiciPath,'auxiliary'));
    addpath(fullfile(amiciPath,'auxiliary','datahash'));
%     addpath(fullfile(amiciPath,'@amimodel'));
    addpath(fullfile(amiciPath,'symbolic'));
    cd(modelPath);

**Comment by FFroehlich** [comment]
@ indicates that amimodel is a matlab class (see http://de.mathworks.com/help/matlab/matlab_oop/organizing-classes-in-folders.html for more information about class and path folders and http://de.mathworks.com/help/matlab/object-oriented-programming.html for more general information about classe) 

All functions in AMICI are designed to run independent of the current working directory. This was not properly implemented in the installToolbox case. In the newest revision this should be fixed.

For the problems with the model definitions, the problem was that they were not included in the matlab path. I updated the error message to make this more clear for the user.

I tried to fix all other warnings and errors in the compilation in the newest revision.
